test/lifecycle_module_test.dart

```diff
@@ -948,20 +908,265 @@ void main() {
+                      childModule.eventList,
+                      containsAllInOrder([]
+                        ..addAll([
+                          'onShouldUnload',
+                          'onShouldUnload',
+                          'willUnload',
+                          'onUnload',
+                          'didUnload',
+                        ])
+                        ..addAll(expectedDisposalStates)));
+                }
+              });
+
+              group('with onShouldUnload=false', () {
+                setUp(() {
+                  withChild
+                      ? childModule.mockShouldUnload = false
+                      : module.mockShouldUnload = false;
+                });
+
+                test('should dispose despite the unload being canceled',
+                    () async {
+                  expectInLifecycleState(module, state);
+                  await module.dispose();
+                  expect(module.isDisposed, isTrue);
+                  expect(
+                      module.eventList,
+                      equals([]
+                        ..addAll(expectedPreUnloadStates)
+                        ..addAll(['onShouldUnload'])
+                        ..addAll(expectedDisposalStates)));
+
+                  if (withChild) {
+                    expect(childModule.isDisposed, isTrue);
+                    expect(
+                        childModule.eventList,
+                        containsAllInOrder([]
+                          ..addAll(['onShouldUnload', 'onShouldUnload'])
+                          ..addAll(expectedDisposalStates)));
+                  }
+                });
+
+                test(
+                    'should warn that the unload was canceled but that disposal will continue',
+                    () async {
+                  expect(
+                      Logger.root.onRecord,
+                      emitsThrough(logRecord(
+                          level: Level.WARNING,
+                          message: contains(
+                              '.dispose() was called but Module "${module.name}" canceled'))));
+                  expectInLifecycleState(module, state);
+                  await module.dispose();
+                });
+              });
+
+              group('with an onUnload that throws', () {
+                setUp(() {
+                  withChild
+                      ? childModule.onUnloadError = testError
+                      : module.onUnloadError = testError;
+                });
+
+                test('should emit the unload failure from didUnload', () async {
+                  expect(module.didUnload.first, throwsA(same(testError)));
+                  if (withChild) {
+                    expect(
+                        childModule.didUnload.first, throwsA(same(testError)));
+                  }
+                  expectInLifecycleState(module, state);
+                  await module.dispose();
+                  print('DISPOSED');
```

Comment:
#nit not needed anymore, could be cleaned up on next update. no need to re-review for it